File: examples/upload_image.js
import { S3Client, PutObjectCommand } from "@aws-sdk/client-s3";
import fs from "fs";
import path from "path";

const s3 = new S3Client({
    region: "us-east-1", // MinIO requires a region, though ignored
    endpoint: "https://images.drakernoise.com",
    credentials: {
        accessKeyId: "YOUR_ACCESS_KEY",
        secretAccessKey: "YOUR_SECRET_KEY"
    }
});

async function uploadImage(filePath) {
    const fileName = path.basename(filePath);
    const fileContent = fs.readFileSync(filePath);

    const command = new PutObjectCommand({
        Bucket: "public-assets",
        Key: fileName,
        Body: fileContent
    });

    try {
        await s3.send(command);
        const url = `https://images.drakernoise.com/public-assets/${fileName}`;
        console.log(`Successfully uploaded: ${url}`);
        return url;
    } catch (err) {
        console.error("Error uploading file:", err);
    }
}

// Usage
uploadImage("./test_image.jpg");

File: examples/upload_image.py
import boto3
from botocore.client import Config
import os

# Configuration
ENDPOINT = 'https://images.drakernoise.com'
ACCESS_KEY = 'YOUR_ACCESS_KEY'
SECRET_KEY = 'YOUR_SECRET_KEY'
BUCKET_NAME = 'public-assets'

def upload_image(file_path):
    s3 = boto3.client('s3',
        endpoint_url=ENDPOINT,
        aws_access_key_id=ACCESS_KEY,
        aws_secret_access_key=SECRET_KEY,
        config=Config(signature_version='s3v4')
    )

    file_name = os.path.basename(file_path)
    
    try:
        # Upload file
        s3.upload_file(file_path, BUCKET_NAME, file_name)
        
        # Generate Public URL
        url = f"{ENDPOINT}/{BUCKET_NAME}/{file_name}"
        print(f"Successfully uploaded: {url}")
        return url
        
    except Exception as e:
        print(f"Error uploading file: {e}")
        return None

if __name__ == "__main__":
    upload_image("test_image.jpg")
